src/base/bittorrent/sessionimpl.cpp

```diff
@@ -3872,6 +3904,86 @@ void SessionImpl::setAdditionalTrackers(const QString &trackers)
     populateAdditionalTrackers();
 }
 
+bool SessionImpl::isAddTrackersFromURLEnabled() const
+{
+    return m_isAddTrackersFromURLEnabled;
+}
+
+void SessionImpl::setAddTrackersFromURLEnabled(bool enabled)
+{
+    if (enabled != isAddTrackersFromURLEnabled())
+    {
+        m_isAddTrackersFromURLEnabled = enabled;
+        if (enabled)
+        {
+            m_updateTrackersFromURLTimer->start();
+            updateTrackersFromURL();
+        }
+        else
+        {
+            m_updateTrackersFromURLTimer->stop();
+            setAdditionalTrackersFromURL(u""_s);
+        }
+    }
+}
+
+QString SessionImpl::additionalTrackersURL() const
+{
+    return m_additionalTrackersURL;
+}
+
+void SessionImpl::setAdditionalTrackersURL(const QString &url)
+{
+    if (url != additionalTrackersURL())
+    {
+        m_additionalTrackersURL = url.trimmed();
+        if (isAddTrackersFromURLEnabled())
+            updateTrackersFromURL();
+    }
+}
+
+QString SessionImpl::additionalTrackersFromURL() const
+{
+    return m_additionalTrackersFromURL;
+}
+
+void SessionImpl::setAdditionalTrackersFromURL(const QString &trackers)
+{
+    if (trackers != additionalTrackersFromURL())
+    {
+        m_additionalTrackersFromURL = trackers;
+        populateAdditionalTrackersFromURL();
+    }
+}
+
+void SessionImpl::updateTrackersFromURL()
+{
+    const QString &url = additionalTrackersURL();
+    if (url.isEmpty())
+    {
+        setAdditionalTrackersFromURL(u""_s);
+    }
+    else
+    {
+        Preferences *const pref = Preferences::instance();
+        Net::DownloadManager::instance()->download(Net::DownloadRequest(url)
+                , pref->useProxyForGeneralPurposes(), this, &SessionImpl::handleTrackersFromURLDownloadFinished);
+    }
+}
+
+void SessionImpl::handleTrackersFromURLDownloadFinished(const Net::DownloadResult &result)
+{
+    if (result.status == Net::DownloadStatus::Success)
+    {
+        setAdditionalTrackersFromURL(QString::fromUtf8(result.data));
+        LogMsg(tr("Public tracker list updated"), Log::INFO);
```

Comment:
Is it correct to call it "public"?

Reply:
Yes, that's correct.

Reply:
This feature only applies to public (non-private) torrents and is referred to as "public" in qBittorrent Enhanced Edition, but I've moved away from that naming. Just forgot to update it here.